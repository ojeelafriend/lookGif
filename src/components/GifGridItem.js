import React from 'react';
import PropTypes from 'prop-types';

export const GifGridItem = ({ title, gif }) => {
	return (
		<li className="animate__animated animate__bounces animate__bounceInDown">
			<img src={gif} alt={title} />
			<div>
				<h3>{title}</h3>
			</div>
		</li>
	);
};

GifGridItem.propTypes = {
	title: PropTypes.string.isRequired,
	gif: PropTypes.string.isRequired,
};

export default GifGridItem;
