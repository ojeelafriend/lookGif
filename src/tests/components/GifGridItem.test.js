import { shallow } from 'enzyme';

import GifGridItem from '../../components/GifGridItem';

describe('Test <GifGridItem/>', () => {
	const image = {
		id: '3o7aTqfmFq4WwrKl7W',
		title: 'dancers choreo GIF by Justin',
		gif: 'https://media0.giphy.com/media/3o7aTqfmFq4WwrKl7W/giphy.gif?cid=2f7fc425myem6fm8b4k2mru76gfrci0992zl5tpjsl6hsjfk&rid=giphy.gif&ct=g',
	};

	let wrapper;
	beforeEach(() => {
		wrapper = shallow(<GifGridItem title={image.title} gif={image.gif} />);
	});

	test('Snapshot the initial component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('Render component with predetermined props: title, gif', () => {
		const img = wrapper.find('img');
		expect(img.prop('src')).toBe(image.gif);
		expect(img.prop('alt')).toBe(image.title);

		const title = wrapper.find('h3').text();
		expect(title).toBe(image.title);
	});

	test('Show animation: animate__bounceInDown', () => {
		expect(wrapper.find('li').hasClass('animate__bounceInDown')).toBe(true);
	});
});
