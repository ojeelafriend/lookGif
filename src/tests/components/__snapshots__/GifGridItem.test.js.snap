// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem/> success render Snapshot the initial component 1`] = `
<li
  className="animate__animated animate__bounces animate__bounceInDown"
>
  <img
    alt="dancers choreo GIF by Justin"
    src="https://media0.giphy.com/media/3o7aTqfmFq4WwrKl7W/giphy.gif?cid=2f7fc425myem6fm8b4k2mru76gfrci0992zl5tpjsl6hsjfk&rid=giphy.gif&ct=g"
  />
  <div>
    <h3>
      dancers choreo GIF by Justin
    </h3>
  </div>
</li>
`;
